internal/worker/api/errors.go

```diff
@@ -168,22 +180,22 @@ func HTTPErrorHandler(echoError error, c echo.Context) {
 				err = c.JSON(sec.httpStatus, apiErr)
 			}
 			if err != nil {
-				c.Logger().Error(err)
+				c.Logger().Errorf("Failed to return error response: %v", err)
 			}
 		}
 	}
 
 	he, ok := echoError.(*echo.HTTPError)
 	if !ok {
-		doResponse(ErrorNotHTTPError, c, echoError)
+		doResponse(ErrorNotHTTPError, c)
 		return
 	}
 
 	sec, ok := he.Message.(ServiceErrorCode)
 	if !ok {
 		// No service code was set, so Echo threw this error
-		doResponse(apiErrorFromEchoError(he), c, he.Internal)
+		doResponse(apiErrorFromEchoError(he), c)
 		return
 	}
-	doResponse(sec, c, nil)
+	doResponse(sec, c)
```

Comment:
How about changing this to `doResponse(sec, c, he.Internal)` and keeping doResponse's signature as is?